### ISSUE TITLE
Problems running app on Ubuntu 22.10

### ISSUE THREAD

**Issue by MatejKovacic** [barracuda-fsh/pyobd#11]
I have Ubuntu 22.10 and have installed application. I did `git clone`, installed all dependencies, added user to a required group, and from the _pyodb directory_ tried to run an app.

When I try to run it (`python3 pyobd.py`) I get this error:
```

Traceback (most recent call last):
  File "/home/matej/.local/lib/python3.10/site-packages/pint/compat/__init__.py", line 64, in <module>
    from collections import Chainmap
ImportError: cannot import name 'Chainmap' from 'collections' (/usr/lib/python3.10/collections/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/matej/pyobd/pyobd.py", line 53, in <module>
    import obd_io  # OBD2 funcs
  File "/home/matej/pyobd/obd_io.py", line 39, in <module>
    import obd
  File "/home/matej/pyobd/obd/__init__.py", line 41, in <module>
    from .obd import OBD
  File "/home/matej/pyobd/obd/obd.py", line 38, in <module>
    from .commands import commands
  File "/home/matej/pyobd/obd/commands.py", line 36, in <module>
    from .decoders import *
  File "/home/matej/pyobd/obd/decoders.py", line 38, in <module>
    from .UnitsAndScaling import Unit, UAS_IDS
  File "/home/matej/pyobd/obd/UnitsAndScaling.py", line 33, in <module>
    import pint
  File "/home/matej/.local/lib/python3.10/site-packages/pint/__init__.py", line 19, in <module>
    from .unit import (UnitRegistry, LazyRegistry)
  File "/home/matej/.local/lib/python3.10/site-packages/pint/unit.py", line 27, in <module>
    from . import registry_helpers
  File "/home/matej/.local/lib/python3.10/site-packages/pint/registry_helpers.py", line 14, in <module>
    from .compat import string_types, zip_longest
  File "/home/matej/.local/lib/python3.10/site-packages/pint/compat/__init__.py", line 66, in <module>
    from .chainmap import ChainMap
  File "/home/matej/.local/lib/python3.10/site-packages/pint/compat/chainmap.py", line 16, in <module>
    from collections import MutableMapping
ImportError: cannot import name 'MutableMapping' from 'collections' (/usr/lib/python3.10/collections/__init__.py)
```

I also tried to run binary file (`./pyobd1.14linux`) but I get this error:
```
(pyobd1.14linux:12344): Gtk-WARNING **: 14:14:59.987: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Prekinjen (izpis jedra)
```
I tried this:
```
sudo update-mime-database /usr/share/mime
sudo apt-get install shared-mime-info
unset GDK_PIXBUF_MODULEDIR
unset GDK_PIXBUF_MODULE_FILE
sudo /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/gdk-pixbuf-query-loaders --update-cache
rm ~/.cache/icon-cache.kcache
```
...but it is the same.

Also, `bluez-utils` does not exists in Ubuntu 22.10, but it is replaced by `bluez`.

**Comment by rana-ans** [comment]
Hi @MatejKovacic , I'm not sure if @barracuda-fsh has directly imprted from "collections" or not but since you have the latest Linux and Python versions, if you could roll-back to an earlier version of Python (3.9) it will work just fine and if you do not want to that then you may open that specific file and just add ".abc" at the end of "collections", 
so it will be;
`from collections.abc import Chainmap`
and for the next exception
`from collections.abc import MutableMapping`

I hope that helps.

**Comment by MatejKovacic** [comment]
I am not sure which file should that be?

- `/home/matej/.local/lib/python3.10/site-packages/pint/compat/__init__.py` 
- `/usr/lib/python3.10/collections/__init__.py`

So here should I add `try ... except` for `Chainmap` and `MutableMapping`?

**Comment by kalvdans** [comment]
A `pip install --upgrade pint` fixed the `collections` trouble for me (got Pint 0.20.1). Should I make a pull request for that change in `requirements.txt`?

**Comment by MatejKovacic** [comment]
`pip3 install --upgrade pint` did a trick for me.

I also paired VLINK device via GUI (you can see the picture below), but when I try to connect it through the app (`OBD-II` - `Connect`), connection was not successful.

![V-LINK_BT](https://user-images.githubusercontent.com/3339198/233729730-8c2751ea-0046-44a6-9f4c-ea02077f98c0.png)

Do I need to do something additional to pair the V-LINK device? I would say no, but have no idea why does it not connects...

**Comment by MatejKovacic** [comment]
I am attaching LOG file...
[python3 pyobd py ERROR.txt](https://github.com/barracuda-fsh/pyobd/files/11299098/python3.pyobd.py.ERROR.txt)

**Comment by MatejKovacic** [comment]
OK, I found out... I needed to click to an Bluetooth icon in the status bar and then connect V-LINK device (before that I just paired it).

Select `Devices` via BT menu from the status bar: 
![BT_connect](https://user-images.githubusercontent.com/3339198/233734197-2111a4e1-4666-43f7-8e79-9fa424cb32a0.png)

Connect V-LINK:
![V-LINK_connected](https://user-images.githubusercontent.com/3339198/233734214-773eb29e-d286-44d1-91d6-fa78a8ee67e1.png)

Finally, I needed to connect OBDII through the app, and got the data!

**Comment by barracuda-fsh** [comment]
Yes, you have to pair AND connect bluetooth device if you are using bluetooth device.

I updated the requirements.txt file so that there is no specific version now, which means that it will install the latest by default. That should do it. Pint was specified the older version only because that was specified in the python-obd library, which was not updated since 2019, but I have not found it to not work with the latest version, that's because it does work with the latest version.

In short, I updated requierements.txt. I will also have to build the new executable files to avoid people having problems. Will do ASAP.
